azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/GenericResource.java

```diff
@@ -140,10 +159,62 @@
     }
 
     /**
+     * A generic resource update allowing to change the resource properties.
+     */
+    interface UpdateWithProperties {
+        /**
+         * Specifies other properties of the resource.
+         *
+         * @param properties the properties object
+         * @return the next stage of generic resource update
+         */
+        Update withProperties(Object properties);
+    }
+
+    /**
+     * A generic resource update allowing to change the parent resource.
+     */
+    interface UpdateWithParentResource {
+        /**
+         * Specifies the parent resource.
+         *
+         * @param parentResourceId the UUID of the parent resource
+         * @return the next stage of the generic resource definition
+         */
+        Update withParentResource(String parentResourceId);
+    }
+
+    /**
+     * A generic resource update allowing to change the resource plan.
+     */
+    interface UpdateWithPlan {
+        /**
+         * Specifies the plan of the resource.
+         *
+         * @param name the name of the plan
+         * @param publisher the publisher of the plan
+         * @param product the name of the product
+         * @param promotionCode the promotion code, if any
+         * @return the next stage of the generic resource update
+         */
+        Update withPlan(String name, String publisher, String product, String promotionCode);
+
+        /**
+         * Specifies the plan of the resource.
+         *
```

Comment:
I actually don't quite understand what this plan is. Tried it out but Azure told me only 3rd party apps can have plans. Since it's optional, I just exposed these 2 options.